### PR TITLE
Added ad4630 support

### PR DESCRIPTION
**Pull request by darshiltrivedi**
# Description

Added Initial support of ad4630-24 & ad4030-24. pyadi device interface, examples, device doc and hardware test was added.

Fixes # (issue)
Previous developer version i.e. branch ad463x(that was never added to main) did not have channel attributes and was not compatible with ad4030-24

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] pytest hardware test - dma_rx test and multiple values attribute test

**Test Configuration**:
* Hardware: Physcially Eval-ad4630 and EVal-4030 were tested to produce result equivalent to lab test/graphs.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

@mthoren-adi

### CODE REVIEW

File: adi/ad4630.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                _channels.append((ch.id, self._diff_channel(self._ctrl, ch.id)))
+                if "0" in ch.name:
+                    self.chan0 = self._diff_channel(self._ctrl, ch.name)
+                if "1" in ch.name:
+                    self.chan1 = self._diff_channel(self._ctrl, ch.name)
+                continue
+
+        rx.__init__(self)
+
+    def rx(self):
```
Comment (tfcollins) [inline adi/ad4630.py:97]: On the rx method, could you use the existing shift and masking features already built in? Do all channels have unique bits and shifts?
This classes uses the mask+shifts https://github.com/analogdevicesinc/pyadi-iio/blob/master/adi/ad7124.py

File: adi/ad4630.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        for ch in self._ctrl.channels:
+            self.output_bits.append(ch.data_format.bits)
+            self._rx_channel_names.append(ch.id)
+            if "differential" in ch.name:
+                _channels.append((ch.id, self._diff_channel(self._ctrl, ch.id)))
+                if "0" in ch.name:
+                    self.chan0 = self._diff_channel(self._ctrl, ch.name)
+                if "1" in ch.name:
+                    self.chan1 = self._diff_channel(self._ctrl, ch.name)
+                continue
```
Comment (tfcollins) [inline adi/ad4630.py:93]: Unnecessary continue

File: adi/ad4630.py (hunk 1 above)
Comment (darshiltrivedi) [reply adi/ad4630.py:97]: All the channels may have unique bits and shift, mask but different output data mode have different bits and shift so single mask value will not work.

File: adi/ad4630.py (hunk 2 above)
Comment (darshiltrivedi) [reply adi/ad4630.py:93]: Sorry. I will remove it along with any other changes required to this file. If not I can do it.

**Pull request merged**